### ISSUE TITLE
'xeus C++11' kernel not appear in the jupyter notebook menu.

### ISSUE THREAD

**Issue by flashlib** [QuantStack/xeus-cling#104]
Hi, 
I installed the xeus-cling followed by the instruction of following:
```
conda create -n cling
source activate cling

conda install cling -c QuantStack -c conda-forge
conda install xeus-cling -c QuantStack -c conda-forge
conda install notebook -c conda-forge
```

But the 'xeus C++11' kernel does not appear in the jupyter notebook menu.

Here is some debug informations:
```
(cling) root@7bac7e6baf53:/notebooks# which jupyter
/root/miniconda3/bin/jupyter
(cling) root@7bac7e6baf53:/notebooks# ll /root/miniconda3/envs/cling/share/jupyter/kernels/
total 16
drwxr-xr-x 4 root root 4096 Feb  1 22:45 ./
drwxr-xr-x 4 root root 4096 Feb  1 22:45 ../
drwxr-xr-x 2 root root 4096 Feb  1 22:45 xeus-cling-cpp11/
drwxr-xr-x 2 root root 4096 Feb  1 22:45 xeus-cling-cpp14/
```
Can anyone tell me why?
Thanks very much!

**Comment by flashlib** [comment]
More info:
```
root@7bac7e6baf53:~/.jupyter# jupyter kernelspec list
Available kernels:
  python3    /root/miniconda3/share/jupyter/kernels/python3
  python2    /usr/local/share/jupyter/kernels/python2
root@7bac7e6baf53:~/.jupyter# which jupyter
/root/miniconda3/bin/jupyter
root@7bac7e6baf53:~/.jupyter# source activate cling
(cling) root@7bac7e6baf53:~/.jupyter# which jupyter
/root/miniconda3/bin/jupyter
(cling) root@7bac7e6baf53:~/.jupyter# jupyter kernelspec list
Available kernels:
  python3    /root/miniconda3/share/jupyter/kernels/python3
  python2    /usr/local/share/jupyter/kernels/python2
(cling) root@7bac7e6baf53:~/.jupyter#
```

**Comment by SylvainCorlay** [comment]
The kernel specs appear to be installed in `cling` environment. Can you try launching notebook from an activated cling environment?

**Comment by gouarin** [comment]
I also have a question about your installation: why did you install miniconda with the root user and not in your home directory ?

**Comment by flashlib** [comment]
It's the same. I think it is because there is a jupyter notebook on my system before install conda. I am going to install on a clear docker image. I will feedback later. Thanks.

**Comment by flashlib** [comment]
I reinstalled on ubuntu and centos docker image but meet the error: "post-link script failed for package QuantStack::gcc-7-7.2.0-1".
I finally installed on MAC os and it run.

**Comment by SylvainCorlay** [comment]
The post-link issue is something that has been reported earlier and which we are not reproducing locally.

In fact we only depend on gcc's STL so we should remove the faulty postlink check fron the recipe as a workaround.

**Comment by aforechi** [comment]
Dear @SylvainCorlay, I'm facing this post-link issue while trying to deploy a xtensor notebook from quantstack repo using mybinder.org. In order to reproduce it, one could deploy a new jupyter server for xtensor repo through mybinder.org. I hope this may help in reproducing the post-link issue.

**Comment by SylvainCorlay** [comment]
@aforechi I am not sure I understand. Do you have a local deployment of mybinder ?

**Comment by aforechi** [comment]
No. I refer to [https://mybinder.org](https://mybinder.org). There one can easily deploy jupyter notebook servers just by filling a form with a github repo such as for xtensor. By doing so we'll get the following [link](https://mybinder.org/v2/gh/QuantStack/xtensor/master?filepath=notebooks/xtensor.ipynb). Following the link you can read the post-link error messages.

**Comment by SylvainCorlay** [comment]
Ok gotcha.

We provide a binder link in the README (both as a badge and later with the
binder logo) which are functional.

I am AFK at the moment but will check the logs later.

S.

**Comment by aforechi** [comment]
Great @SylvainCorlay! Actually, I'm facing the same issue but with my own [repo](https://github.com/aforechi/ifes-cpp), which I use for teaching C++ programming. I must thank you all at QuantStack for making this more interesting! I referred earlier to the xtensor repo because the issue might be correlated. Cheers!

**Comment by gouarin** [comment]
This issue happens when the default gcc compiler is too old.

Could you try to install the last gcc compiler by using the apt.txt file in
binder and see if the problem is solved?

**Comment by SylvainCorlay** [comment]
@aforechi OK, I think I might have a fix. Stay tuned.

**Comment by SylvainCorlay** [comment]
This is fixed now!

I am closing the issue but don't hesitate to comment here should you have the issue again and I will reopen.

**Comment by j-keiter** [comment]
I do not see any C++ options when running jupyter-notebook with google chrome.  Everything seems to work.  I activate the (xeus-cling) environment and launch jupyter-notebook.  
conda 4.6.8
Linux 4.19.0-2-amd64 #1 SMP Debian 4.19.16-1 (2019-01-17) x86_64 GNU/Linux
binutils_impl_linux-64    2.31.1               h6176602_1  
binutils_linux-64         2.31.1               h6176602_3  
clang_variant             1.0               cling_6.14.06    conda-forge
clangdev                  5.0.0             h935a590_1004    conda-forge
cling                     0.5               he860b03_1006    conda-forge
cppzmq                    4.3.0             h484f380_1000    conda-forge
cryptopp                  7.0.0             hf484d3e_1003    conda-forge
gcc_impl_linux-64         7.3.0                habb00fd_1    conda-forge
gcc_linux-64              7.3.0                h553295d_3    conda-forge
gxx_impl_linux-64         7.3.0                hdf63c60_1    conda-forge
gxx_linux-64              7.3.0                h553295d_3    conda-forge
libgcc-ng                 8.2.0                hdf63c60_1  
libsodium                 1.0.16            h14c3975_1001    conda-forge
libstdcxx-ng              8.2.0                hdf63c60_1  
libuuid                   2.32.1            h14c3975_1000    conda-forge
nlohmann_json             3.4.0                hf484d3e_0    conda-forge
xeus                      0.18.1            h6e680fc_1000    conda-forge
xeus-cling                0.4.11               he860b03_0    conda-forge
xtl                       0.5.4                h6bb024c_0    conda-forge
zeromq                    4.2.5             hf484d3e_1006    conda-forge
zlib                      1.2.11            h14c3975_1004    conda-forge

**Comment by pythologist** [comment]
Hi. 

Could this issue not be closed until we know at least which version of gcc must be installed? 

If that is indeed the issue - and not "I installed jupyter before I installed conda" if that is the issue?

By the way The "I installed jupyter" issue stems from having met some people who wanted to use Jupyter without using conda. And if that is the issue, then it would be good to inform people about this.

Thanks!

Thanks
